chargebee/src/main/java/com/chargebee/android/billingservice/BillingClientManager.kt

```diff
@@ -253,6 +255,102 @@ class BillingClientManager(context: Context) : PurchasesUpdatedListener {
 
     }
 
+    internal fun changeProduct(
+        changeProductParams: ChangeProductParams,
+        purchaseCallBack: CBCallback.PurchaseCallback<String>
+    ) {
+        this.purchaseCallBack = purchaseCallBack
+        onConnected({ status ->
+            if (status) {
+                changeProduct(changeProductParams)
+            } else
+                purchaseCallBack.onError(
+                    connectionError
+                )
+        }, { error ->
+            purchaseCallBack.onError(error)
+        })
+
+    }
+
+    private fun changeProduct(changeProductParams: ChangeProductParams) {
+        this.changeProductParams = changeProductParams
+        this.purchaseProductParams = changeProductParams.purchaseProductParams
+        val offerToken = changeProductParams.purchaseProductParams.offerToken
+        val oldProductId = changeProductParams.oldProductId
+        val prorationMode = changeProductParams.prorationMode ?: BillingFlowParams.ProrationMode.IMMEDIATE_WITH_TIME_PRORATION
```

Comment:
Let's start with support for `IMMEDIATE_WITH_TIME_PRORATION`. We would need to build in support for other proration modes in the backend.

Reply:
Done